src/DIRAC/FrameworkSystem/Client/TokenManagerClient.py

```diff
@@ -86,5 +86,7 @@ def getToken(
                 idpObj.revokeToken(token["refresh_token"])
 
         return self.executeRPC(
-            username, userGroup, scope, audience, identityProvider, requiredTimeLeft, call="getToken"
+            userName, userGroup, scope, audience, identityProvider, requiredTimeLeft, call="getToken"
         )
+
+gTokenManager = TokenManagerClient()
```

Comment:
Is it really needed?

Reply:
It is similar to ProxyManagerClient. It should be a global object to keep the cache of already available tokens across multiple threads for example